File: app/src/main/java/edu/umsl/bekzod_tolipov/QuizModel.java
package edu.umsl.bekzod_tolipov;

import java.util.ArrayList;

public class QuizModel {

    private ArrayList<QuizQuestion> mQuestions;
    private Integer currentIndex = 0;

    public QuizModel(Integer index){
        mQuestions = new ArrayList<>();
        mQuestions.add(new QuizQuestion(R.string.question_brazil, true));
        mQuestions.add(new QuizQuestion(R.string.question_einstein, true));
        mQuestions.add(new QuizQuestion(R.string.question_georgia, false));
        mQuestions.add(new QuizQuestion(R.string.question_libya, false));
        mQuestions.add(new QuizQuestion(R.string.question_koalas, true));

        this.currentIndex = index;
    }

    QuizQuestion getCurrentQuestion(){
        return  mQuestions.get(currentIndex);
    }

    public void advanceToNextQuestion(){
        currentIndex = (currentIndex + 1) % mQuestions.size();
        getCurrentQuestion();
    }

    public Integer getCurrentIndex(){
        return currentIndex;
    }

    boolean isAnswerCorrect( boolean answerValue){
        boolean isTrue = mQuestions.get(currentIndex).getIsTrue();
        return isTrue == answerValue;
    }

}

File: app/src/main/java/edu/umsl/bekzod_tolipov/QuizModel.kt
package edu.umsl.bekzod_tolipov

class QuizModel(index: Int) {

    var currentIndex: Int = index
        private set

    var mQuestions: Array<QuizQuestion> = arrayOf(
            QuizQuestion(id = R.string.question_brazil, isTrue = true),
            QuizQuestion(R.string.question_einstein, true),
            QuizQuestion(R.string.question_georgia, false),
            QuizQuestion(R.string.question_libya, false)
    )

    val currentQuestion: QuizQuestion
        get() = mQuestions[currentIndex]

    fun advanceToNextQuestion():Unit{
        currentIndex = (currentIndex + 1) % mQuestions.size
        currentQuestion
    }

    fun isAnswerCorrect(answerValue: Boolean):Boolean{
        val currentQuestion = this.currentQuestion
        return currentQuestion.isTrue == answerValue
    }


}